### PR TITLE
Egoscale V3 - Zone wrapper

### PR DESCRIPTION
**Pull request by kobajagi**
This PR adds zone-aware wrapper around V3 Client.

### CODE REVIEW

**Review by PhilippeChepy — APPROVED** [review]

File: v3/zoned_client.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		oapi.AtVie2,
+		oapi.BgSof1,
+		oapi.ChDk2,
+		oapi.DeFra1,
+		oapi.DeMuc1,
+	}
+)
+
+// ZonedClient is a Exoscale API Client that can communicate with API servers in different zones.
+// It has the same interface as Client and uses currently seleced zone to run API calls.
```
Comment (jessicatoscani) [inline v3/zoned_client.go:31]:
```suggestion
// It has the same interface as Client and uses currently selected zone to run API calls.
```

**Review by jessicatoscani — APPROVED** [review]

**Comment by kobajagi** [comment]
Last commit adds support for env var overrides.

**Review by PhilippeChepy — APPROVED** [review]

File: v3/zoned_client.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+const (
+	DefaultHostPattern = "https://api-%s.exoscale.com/v2"
+
+	EnvKeyAPIEndpointPattern = "EXOSCALE_API_ENDPOINT_PATTERN"
+	EnvKeyAPIEndpointZones   = "EXOSCALE_API_ENDPOINT_ZONES"
+)
+
+var (
+	// DefaultZones list (available in oapi code).
+	// When new zone is added or existing removed this slice needs to be updated.
```
Comment (sauterp) [inline v3/zoned_client.go:22]: I wish we could generate this somehow. Is there really no way?

**Review by sauterp — APPROVED** [review]

File: v3/zoned_client.go (hunk 2 above)
Comment (kobajagi) [reply v3/zoned_client.go:22]: We can generate it, it should be fairly straightforward. I'll create a separate story for that.

**Pull request merged**